internal/testcontext/compile.go

```diff
@@ -29,3 +32,68 @@ func (ctx *Context) Compile(pkg string) string {
 
 	return exe
 }
+
+// CompileShared compiles pkg as c-shared.
+func (ctx *Context) CompileShared(name string, pkg string) Include {
+	ctx.test.Helper()
+
+	base := ctx.File("build", name)
+
+	// not using race detector for c-shared
+	cmd := exec.Command("go", "build", "-buildmode", "c-shared", "-o", base+".so", pkg)
+	ctx.test.Log("exec:", cmd.Args)
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		ctx.test.Error(string(out))
+		ctx.test.Fatal(err)
+	}
+	ctx.test.Log(string(out))
+
+	return Include{Header: base + ".h", Library: base + ".so"}
+}
+
+// CompileC compiles file as with gcc and adds the includes.
+func (ctx *Context) CompileC(file string, includes ...Include) string {
+	ctx.test.Helper()
+
+	exe := ctx.File("build", filepath.Base(file)+".exe")
+
+	var args = []string{}
+	args = append(args, "-ggdb", "-Wall")
```

Comment:
why not just initialize args with these values ?

Reply:
Easier to move the argument order around and add things above and below. But not feeling too strongly in any particular way.